test/e2e/mob_test.go

```diff
@@ -115,23 +115,10 @@ func TestMachineOSBuilder(t *testing.T) {
 
 	t.Cleanup(createConfigMapForTest(t, cs))
 
-	// get the feature gates because we're gating this for now
-	featureGates, err := cs.ConfigV1Interface.FeatureGates().Get(context.TODO(), "cluster", metav1.GetOptions{})
+	featureGateEnabled, err := helpers.CheckFeatureGateEnabled(cs, "OnClusterBuild")
 	require.NoError(t, err, "Failed to retrieve feature gates")
 
-	// TODO(jkyros): this should be a helper or we should use whatever the "best practice" way is
-	// for retrieving the gates during a test, but this works for now
-	var featureGateEnabled bool
-	for _, featureGateDetails := range featureGates.Status.FeatureGates {
-		for _, enabled := range featureGateDetails.Enabled {
-			if enabled.Name == "OnClusterBuild" {
-				featureGateEnabled = true
-
-			}
-		}
-	}
-
-	t.Logf("Feature gate OnClusterBuiild enabled: %t", featureGateEnabled)
+	t.Logf("Feature gate OnClusterBuild enabled: %t", featureGateEnabled)
```

Comment:
suggestion (non-blocking): Call `t.Fatalf()` if the feature gates are not enabled. This way we get a failing test right away with a clear failure message as opposed to it failing in a cryptic way.

Reply:
So I originally had this helper in `msbic_test` when my test was part of the main e2e(this mob_test is still part of e2e - so I left the new helper in there). The `mob_test` also has a path that runs when the feature gates are disabled, where it tests to make sure the MOB isn't active I think. So we'd have to take that bit of the test out if we want to fatally end the test on lack of feature gates. I figured I would let Dalia know once this techpreview suite landed so she can break out the tests as necessary. Sorry about the confusion, I should've mentioned it in the description!

Reply:
My initial opinion is that we can probably remove the portion of the `mob_test` test that requires the featuregate to be disabled since all it does is throw an error and fail (or at least that's what _should_ happen; I'm not sure if the `e2e-gcp-op` job has the featuregates enabled). However, removing code block would require us to move that test into the `e2e-techpreview` suite (though execution order would matter in this case, which isn't great).

Initially, I thought there might be some functional overlap between `mob_test.go` and `onclusterbuild_test.go`, but @cdoern added some additional test cases to `mob_test.go` that deals with restarting the MOB in response to ConfigMap changes (which is fine and worth keeping around, in my opinion).

Here are a couple of additional ideas for things we can do here:
1. We can make that check look for multiple featuregates to be enabled, e.g., `helpers.AssertFeatureGatesEnabled(t, []string{"FeatureGate1", "FeatureGate2"})`.
2. For everything in the `e2e-techpreview` folder, we could create a `TestMain()` function (more on that here: https://medium.com/goingogo/why-use-testmain-for-testing-in-go-dafb52b406bc) that performs the featuregate check before running any of the tests. That would allow us to apply that to all of the tests.

Regardless, we're probably splitting hairs here and I don't want to overcomplicate things :smile:.

Reply:
Hmm, so few things:

-  `e2e-gcp-op` does not have tech-preview enabled, hence `mob_test` has not been actually running the way we hoped since it merged(or atleast the TP gated bit hasn't). The non-TP section of that code does have a `require.Error()` on the MOB deployment get while FG is disabled...although it seems to skip the NotFound error? A little confusing. 
- I'm leaning towards letting that author(Charlie/Dalia's) do the move under a separate PR - this PR is already getting slightly heavy for my liking.  Not strongly held though 
- I'm +1 on both (1) and (2)  having an `AssertFeatureGatesEnabled` in a `TestMain()` makes a lot of sense (: Calling AssertFailure while mob_test is in `e2e-gcp-op` will cause it to permafail, so we'd have to leave it untouched as it was for now. 
- And...Sergio pointed out to me that the CVO automatically restores the configmap so I'll need to make a slight modification to the test 
Let me know what you think!

Reply:
I agree with everything you said. I think the best path forward will be:

- I'll take a look at `mob_test.go` independently of this effort because something clearly isn't right there and it's not on you to fix it :smile:.
- Moving `mob_test.go` into `e2e-techpreview` is also not on you either. Charlie / Dalia can do that when they're ready and once I figure out what's wrong with that test.
- Adding those assertions / helpers is fine. We can add `TestMain` to `e2e-techpreview` for now or if you want to be more granular and do it on a per-test basis, that's fine too.
- Make whatever modifications you need.

Reply:
Sounds good, I'll get to work on this (: